tee-worker/litentry/core/stf-task/receiver/src/lib.rs

```diff
@@ -136,6 +137,10 @@ where
 			return Ok(())
 		}
 
+		// swap the hash in the rpc connection registry to make sure furthre RPC responses go to
+		// the right channel
+		self.author_api.swap_rpc_connection_hash(*old_top_hash, top.hash());
```

Comment:
Have you tested it in multiworker setup? 
I can imagine scenario in which first TOP is processed by worker 1 but second by worker 2. Will rpc client receive correct response ?

According to this comment TOPs might not be synchronized, so maybe this is not a real problem now but may occour in future.
https://github.com/litentry/litentry-parachain/blob/7e096092915036bcff513dbb334528d0a75395f6/tee-worker/sidechain/consensus/aura/src/block_importer.rs#L145 - I couldn't find a code doing sync job.

Reply:
> Have you tested it in multiworker setup?

Nope, but I expect it to work.

Because as mentioned the top pool is **not** synced between workers, so if you submit your request to worker 1 only worker 1 will be able to process it.

Worker 2 will get the processing result as sidechain blocks.

I don't think it will change in the near future

Reply:
It sounds like once we get more workers the time to process this operation will increase linerary and the problem will come back sooner or later.

Reply:
I mean if we have Y workers online then:

- first TOP will be processed in block X
- second TOP will be processed in block X+Y

Reply:
Yes that's a legit assumption. It is out of the scope of this PR but we'll need to address it sooner or later when we start multi-worker setup. We need to have an issue to track this.

II won't have such problem because the tx pool is synced from the parachain.